### PR TITLE
Fix NPE and log location on server start up

### PR DESCRIPTION
**Pull request by benthecarman**
#2160 fixed the NPE on the oracle server start up but broke the log location.

This fixes the NPE in the oracle server start up without breaking the setting of the log location

### CODE REVIEW

File: app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala [hunk 1]
```diff
@@ -80,7 +80,7 @@ trait BitcoinSRunner extends BitcoinSLogger {
   private lazy val network: BitcoinNetwork =
     BitcoinNetworks.fromString(networkStr)
 
-  lazy val datadir: Path = {
+  val datadir: Path = {
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: I think we should keep this lazy. Since we don't actually have test cases for server startup (#2161 ). We have no way to detect regressions on CI. This is one of the joys of Scala's weird behavior with initialization of traits with `val`s inside of it. Sometimes it's too eager and the thing behind the `val` isn't actually filled -- which gives us a `NullPointerException`. 

Also why is logging set here and not somewhere else? Doesn't this need to be done inside every submodule? If not, why isn't logging working as expected in the server project? It seems we have disconnect from library behavior and standalone binary behavior ?

Comment (benthecarman) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: The log file location needs to be set here. We need to parse the config before so we know our datadir and network.

Comment (Christewart) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: >The log file location needs to be set here.  

Why? It seems odd to me that the logging property is set within `val datadir` to begin with.

Comment (benthecarman) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: That is the soonest time we know it.

Comment (Christewart) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: Here is the scala documentation on this situation: https://docs.scala-lang.org/tutorials/FAQ/initialization-order.html

TLDR: Use `lazy val`

I was actually unaware of this compiler flag: `-Xcheckinit`. Thats pretty cool.

Comment (benthecarman) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: I guess we can move it back to `lazy val`, we just then lose our guarantee that the log file location is set before the application starts

Comment (benthecarman) [reply app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala]: Changed it so it uses a `lazy val` again and will now set the log location automatically before the setting up anything else

**Review by Christewart — APPROVED** [review]

**Comment by Christewart** [comment]
Approved, but can you do one last sanity check locally that everything is working with appSEver and oracleServer?

**Comment by benthecarman** [comment]
> Approved, but can you do one last sanity check locally that everything is working with appSEver and oracleServer?

Good call, needed a small fix for the oracle, all good now!

**Comment by Christewart** [comment]
> > Approved, but can you do one last sanity check locally that everything is working with appSEver and oracleServer?
> 
> Good call, needed a small fix for the oracle, all good now!

IT's soooooo easy to do! THat's why we need #2161
